### PR TITLE
ref(py3): Remove all py2-compat requirement definitions

### PR DESCRIPTION
**Pull request by BYK**

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -35,7 +31,7 @@ Pillow>=6.2.2,<7.0.0
 progressbar2>=3.32,<3.33
 psycopg2-binary>=2.7.0,<2.9.0
 PyJWT>=1.5.0,<1.6.0
-pyrsistent==0.16.0 # TODO(py3): 0.17.0 requires python3, see https://github.com/tobgu/pyrsistent/issues/205
```
Comment (untitaker) [inline requirements-base.txt:38]: This upgrade seems safe, most are formatting changes, CI changes and removal of six: https://github.com/tobgu/pyrsistent/compare/v0.16.0...v0.17.3

**Review by untitaker — APPROVED** [review]

**Comment by untitaker** [comment]
this should have an accompanying PR in getsentry

File: requirements-base.txt [hunk 2]
```diff
@@ -105,5 +84,4 @@ setproctitle>=1.1.7,<1.2.0
 phabricator>=0.6.0,<0.8
 
 # test dependencies, but unable to move to requirements-test until sentry.utils.pytest and similar are moved to tests/
-mock==3.0.5 ; python_version < "3.3"
```
Comment (joshuarli) [inline requirements-base.txt:108]: @BYK You missed `src/sentry/utils/compat/mock.py`, but I also see a few direct `import mock`s.... how are those not failing? I can fail it locally:

```
tests/sentry/mail/test_adapter.py:7: in <module>
    import mock
E   ModuleNotFoundError: No module named 'mock'
```

Comment (joshuarli) [reply requirements-base.txt:108]: Huh, `mock`'s still being installed in CI. But not the others. It must be a transient dependency, I just directly pip uninstalled mock from my dev env. Lemme try the whole thing.

**Pull request merged**